tracer/src/Datadog.Trace.Tools.dd_dotnet/Program.cs

```diff
@@ -0,0 +1,89 @@
+// <copyright file="Program.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.CommandLine;
+using System.CommandLine.Builder;
+using System.CommandLine.Help;
+using System.CommandLine.Parsing;
+using System.Runtime.InteropServices;
+
+namespace Datadog.Trace.Tools.dd_dotnet;
+
+internal class Program
+{
+    internal static Action<string, string, Dictionary<string, string>?>? CallbackForTests { get; set; }
+
+    internal static int Main(string[] args)
+    {
+        // Disable QUIC support. It requires reflection (and thus increases the final size of the binary)
+        AppContext.SetSwitch("System.Net.SocketsHttpHandler.Http3Support", false);
```

Comment:
Just a thought, but I remember reading in .NET 8 they changed the way compression is implemented in .NET 8 so that if you're not using it, the code paths are trimmable (they aren't in .NET 7). I assume there's nothing we can do about that here, and as we're going to .NET 8 ultimately it probably doesn't matter. Ignore me

Reply:
😛 

HTTP3 is not trimmed even on .NET 8 though. That will be in .NET 9 :(